core/state_processor_rip7560.go

```
@@ -37,9 +37,16 @@ func UnpackValidationData(validationData []byte) (authorizerMagic uint64, validU
 }
 
 func UnpackPaymasterValidationReturn(paymasterValidationReturn []byte) (validationData, context []byte) {
+	if len(paymasterValidationReturn) < 96 {
+		return nil, nil
+	}
 	validationData = paymasterValidationReturn[0:32]
 	//2nd bytes32 is ignored (its an offset value)
 	contextLen := new(big.Int).SetBytes(paymasterValidationReturn[64:96])
+	if uint64(len(paymasterValidationReturn)) < 96+contextLen.Uint64() {
+		return nil, nil
```

Please return an error here. We can't proceed if this fails anyway.